src/re.rs

```diff
@@ -970,11 +970,11 @@ impl<'t> Index<usize> for Captures<'t> {
 ///
 /// # Panics
 /// If there is no group named by the given value.
-impl<'t> Index<&'t str> for Captures<'t> {
+impl<'t, 'i> Index<&'i str> for Captures<'t> {
```

Comment:
Good point; thanks.

While I'm editing the docs: I noticed that the `str` can't outlive the `Captures` if it's obtained through either `Index` impl; should that be a separate PR?  It's a little surprising, so a warning and advice to use methods instead might be nice.  (This is part of the definition of the `Index` trait — normally `a[i]` is part of `a` and can't outlive it — so it can't be fixed without changing the `Output` type to `&'t str` and breaking compatibility.)